app/controllers/gems_adoptions_controller.rb

```diff
@@ -0,0 +1,21 @@
+class GemsAdoptionsController < ApplicationController
+  
```

Comment:
I think you forgot to check if a user is logged-in. We shouldn't allow users that are nor signed-in to access at least `create` and `destroy` actions in this controller.